peercrawler.py

```diff
@@ -12,11 +12,18 @@
 from functools import reduce
 from typing import Iterable, Optional
 
+from pydot import Dot, Node, Edge
+
 import confirm_req
 import telemetry_req
 from msg_handshake import *
+from logger import setup_logging
 
-from pydot import Dot, Node, Edge
+
+LOGGER_NAME = "peercrawler"
+logger = logging.getLogger(LOGGER_NAME)
+if not logger.hasHandlers():  # peercrawler.py is imported from other files, so this check exists to ensure this logger is only configured once
```

Comment:
This should be an assert

Reply:
Since the peercrawler always gets run twice the assertion would always throw. The only solution would be to fix the imports, which would take way longer than just leaving this like this.

Reply:
Why is peercrawler run twice?

Reply:
If you run peercrawler.py:
peercrawler.py L15: `import confirm_req`
confirm_req.py L14: `from peercrawler import *`
Making the peercrawler.py run twice.

Reply:
OK, I see now, the problems are not the imports.
The problem is that the code is in the module level.

Reply:
The peercrawler class should be given a logger to use. And to make it convinient to use, we could make the optional, so that peercrawler creates its own logging, if no logger is given.

Reply:
It is generally not a good idea to have code execute just because you do an import.